File: Benchmarking/30 Euler/euler15/euler15.java
import java.util.*;

public class euler15
{
	public static void main(String[] args)
	{
		int gridSize = Integer.parseInt(args[0]);
		ArrayList<ArrayList<Long>> grid = new ArrayList<ArrayList<Long>>();
		ArrayList<Long> temp;
		for(int i = 0; i < gridSize + 1; i++)
		{
			temp = new ArrayList<Long>();
			for(int j = 0; j < gridSize + 1; j++)
			{
				temp.add(1l);
			}
			grid.add(temp);
		}
		calc(grid, gridSize);
	}
	
	private static long calc(ArrayList<ArrayList<Long>> grid, int gridSize)
	{
		for(int i = gridSize-1; i >= 0; i--)
		{
			for(int j = gridSize-1; j >= 0; j--)
			{
				grid.get(i).set(j, grid.get(i + 1).get(j) + grid.get(i).get(j + 1));
			}
		}
		return grid.get(0).get(0);
	}
}

File: Benchmarking/30 Euler/euler15/euler15.py
import sys

gridSize = int(sys.argv[1])

grid = [ [ 1 for i in range(gridSize+1) ] for j in range(gridSize+1) ]

def calc(grid, gridSize):
  for i in range(gridSize-1, -1, -1):
    for j in range(gridSize-1, -1, -1):
      grid[i][j] = grid[i+1][j] + grid[i][j+1]
  return grid[0][0]

calc(grid, gridSize)
